src/RelayPricer.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi) {
+        this.sourceApi = sourceApi;
+    }
+
+    async getPrice(
+        sourceCurrency: string,
+        targetCurrency: string,
+        intermediateCurrency?: string
+    ): Promise<number> {
+        const intermediary = intermediateCurrency
+            ? intermediateCurrency
+            : 'USD';
+
+        const [sourcePrice, targetPrice] = await Promise.all([
+            this.sourceApi.query(sourceCurrency, intermediary),
+            this.sourceApi.query(targetCurrency, intermediary)
+        ]);
+        if (sourcePrice === 0 || targetPrice === 0) {
+            throw new Error('price cannot be zero');
+        }
+        return sourcePrice / targetPrice;
```

Comment:
I don't get this. The method is getPrice, the test says it should return the currency conversion price, but what it returns is an exchange rate. What am I missing?

Reply:
On my understanding, the currency conversion price is the same as exchange rate, I could rename it.

Reply:
actually, I only just realised what this is doing, lol.
so here we're returning ratio of ratios? I think that's not going to work is it? Isn't it just same as saying for example:
$EUR/JPY == EUR/USD/USD/JPY$

Reply:
renaming but the exchange works like the following:

sourceCurrency=EUR;
targetCurrency=JPY;
intermediateCurrency=USD;

sourceExchangeRate=USD/EUR
targetExchangeRate=USD/JPY

sourceExchangeRate/targetExchangeRate

Reply:
my bad 👍